plasmapy/formulary/tests/test_relativity.py

```diff
@@ -29,6 +29,14 @@ def test_Lorentz_factor():
         Lorentz_factor(3 * u.m / u.s) * u.dimensionless_unscaled
     ).unit == u.dimensionless_unscaled
 
+    assert np.isnan(Lorentz_factor(np.nan * u.m / u.s))
+
+    assert np.all(np.isnan(Lorentz_factor(np.array([np.nan, np.nan]) * u.m / u.s)))
+
+    numerical_result, nan_result = Lorentz_factor(np.array([1, np.nan]) * u.m / u.s)
+    assert np.isnan(nan_result)
+    assert not np.isnan(numerical_result)
+
```

Comment:
```suggestion

    def test_lorentz_factor_nan_input():
        assert np.isnan(Lorentz_factor(np.nan * u.m / u.s))


    def test_lorentz_factor_array_of_nans():
        assert np.all(np.isnan(Lorentz_factor(np.array([np.nan, np.nan]) * u.m / u.s)))


    def test_lorentz_factor_nan_in_array():
        numerical_result, nan_result = Lorentz_factor(np.array([1, np.nan]) * u.m / u.s)
        assert np.isnan(nan_result)
        assert not np.isnan(numerical_result)


    def test_lorentz_factor_exceptions():
```
The tests do look like they're testing the behavior that I was hoping would be added to `Lorentz_factor`, so thank you!

This suggestion breaks up the tests that you added into different functions. A while after we originally wrote the tests for `Lorentz_factor`, we learned that it's a better practice to have each test verify a single unit of behavior.  If there are multiple unrelated `assert` statements in a test (like how the tests of `Lorentz_factor` were written before this pull request), then if one of them fails, the subsequent tests won't get run which could potentially hide errors.